x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/analysis/analyzer/VerifierErrorMessagesTests.java

```diff
@@ -203,6 +203,26 @@ public void testExtractNonDateTime() {
         assertEquals("1:8: Invalid datetime field [ABS]. Use any datetime function.", error("SELECT EXTRACT(ABS FROM date) FROM test"));
     }
 
+    public void testDateTruncInvalidArgs() {
+        assertEquals("1:8: first argument of [DATE_TRUNC(int, date)] must be [string], found value [int] type [integer]",
+            error("SELECT DATE_TRUNC(int, date) FROM test"));
+        assertEquals("1:8: second argument of [DATE_TRUNC(keyword, keyword)] must be [date or datetime], found value [keyword] " +
+                "type [keyword]", error("SELECT DATE_TRUNC(keyword, keyword) FROM test"));
+        assertEquals("1:8: first argument of [DATE_TRUNC('invalid', keyword)] must be one of [MILLENNIUM, CENTURY, DECADE, " + "" +
```

Comment:
It's worth checking whether the message can be improved by using the levenshtein distance to suggest the best appropriate match (did you mean?) same as with do with field names.
It might not work though for short fields so it should be applied for properties with at least 3-4 chars.

Reply:
Nice suggestion, thx! Please check edf42ce75fadd3f99270804dbe47911e88d1e719 in case you see some further improvement there.